### ISSUE TITLE
RepeatVector equivalent in tflearn?

### ISSUE THREAD

**Issue by aryopg** [tflearn/tflearn#662]
I saw several keras discussion in making an attention mechanism, and i saw a line of code that used RepeatVector, is there any equivalent for it in tflearn? or is there any other way to achieve an attention mechanism in tflearn? (I'm trying to make a text classifier with this). Here's the keras code that i talked about : 

`activations = LSTM(units, return_sequences=True)(embedded)

# compute importance for each step
attention = TimeDistributed(Dense(1, activation='tanh'))(activations) 
attention = Flatten()(attention)
attention = Activation('softmax')(attention)
attention = RepeatVector(units)(attention)
attention = Permute([2, 1])(attention)`

**Comment by aymericdamien** [comment]
You can have a check to `tflearn.layers.core.time_distributed`

**Comment by aryopg** [comment]
Oops, i've copied the code badly. Here's the code:
```
attention = TimeDistributed(Dense(1, activation='tanh'))(activations) 
attention = Flatten()(attention) 
attention = Activation('softmax')(attention) 
attention = RepeatVector(units)(attention) 
attention = Permute([2, 1])(attention)
```

What about the RepeatVector and the Permute part? Is the `tflearn.layers.core.time_distributed` support this? and is there any example to achieve that? The real implementation of attention mech will be great too lol

**Comment by pravinv14** [comment]
It is suggested to use encoder and decoder network for many to many architectures of different input and output length, so can you please provide an example using `RepeatVector` equivalent in TFlearn for this. 

The time dimension from the input is carried till the time_distributed layer, but for many to many architecture of different input and output length. What is the thing to use, to get an output of different length (i.e) of different time steps as per my data set?

**Comment by StealthySemicolon** [comment]
Use This:
`
def repeat(x, n):
    assert ndim(x) == 2
    x = tf.expand_dims(x, 1)
    pattern = tf.stack([1, n, 1])
    return tf.tile(x, pattern)
`
Found this in the keras backend code. This is what the keras repeat vector layer uses in the call function of the layer. x is the input and n is the number of times the tensor has to be repeated. :)

**Comment by StealthySemicolon** [comment]
Sorry for the bad formatting. Here is a link:
https://github.com/keras-team/keras/blob/master/keras/backend/tensorflow_backend.py
Line 2081

**Comment by StealthySemicolon** [comment]
I would also suggest you to put that inside a name scope

**Comment by StealthySemicolon** [comment]
For the ndim, you can find the function in the keras backend for tensorflow. Its the 6th result if you ctrl+f "ndim"
